akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1435,6 +1435,7 @@ private[stream] object Collect {
     private var groupEmitted = true
     private var finished = false
     private var totalWeight = 0L
+    private var elementCount = 0L
```

Comment:
I can replace `elementCount` with a `hasElements: Boolean`... Not much of a difference, but maybe it would be clearer (and a bit more performant...).

Let me know if you want me to do it, or if you find a clever way around this :wink: